### PR TITLE
fix(DataChangeRecord): Workaround nested array limitation with DataDifferencer

### PR DESCRIPTION
**Pull request by silbinarywolf**
WIP - fix(DataChangeRecord): Workaround nested array limitation with DataDifferencer by only json_decode'ing 1 level down. This resolve bugs when you're storing JSON data in a 'Text' DB field via MultiValueField. 

**Test result**
- https://travis-ci.org/symbiote/silverstripe-datachange-tracker/jobs/396219565

**Contains**
- Update Travis CI to actually run tests
- Add functional test to make sure the CMS edit page loads (ie. getCMSFields() has no bugs)
  - (Also tested reintroducing the bug to ensure this is working as expected)
- Add test to ensure custom defined `table_name` classes will track records correctly.
- Removed lots of trailing whitespace at the end of lines. (Was automatically done from Sublime Text 3 IDE + Editorconfig plugin using the library's .editorconfig file)

**Backported here:**
- https://github.com/symbiote/silverstripe-datachange-tracker/releases/tag/3.3.7
- https://github.com/symbiote/silverstripe-datachange-tracker/releases/tag/4.0.6

### CODE REVIEW

File: src/Model/TrackedManyManyList.php [hunk 1]
```diff
@@ -34,27 +36,50 @@ protected function recordManyManyChange($type, $item)
...
+     *
+     * Stripped down version from framework that does not attempt to strip _Live and _versions postfixes as
+     * that throws errors in its preg_match(). (At least it did as of 2018-06-22 on SilverStripe 4.1.1)
+     *
+     * @param string $table
+     * @return string|null The FQN of the class, or null if not found
+     */
+    private function tableClass($table)
+    {
+        $tables = DataObject::getSchema()->getTableNames();
```
Comment (nglasl) [inline src/Model/TrackedManyManyList.php]: How does this work when something isn't using a `table_name` for some reason? It's a little hard to follow the code, but I guess I'm just trying to figure out how this works when a namespace is/isn't present compared to how it worked previously.

Comment (nglasl) [reply src/Model/TrackedManyManyList.php]: I'm assuming this is where the with/without underscores testing comes into play. In which case, I don't see any issues here. Happy for this to be merged and tagged.

Comment (silbinarywolf) [reply src/Model/TrackedManyManyList.php]: The issue with the old code was it tried to figure out the class name from the table name (That's how the SS 3.X code does it). This straight up wouldn't work in some cases. So I fixed the lookup to use core SS functionality.

**Comment by silbinarywolf** [comment]
Tagged in 5.0.2
